less/plugins/breakpoints.js

```diff
@@ -20,27 +20,25 @@ function lookupVariable(context, variableName) {
 	})
 }
 
-function listToMap({ value: list } = { value: [] }) {
+function rulesetToMap({ ruleset: { rules } } = { ruleset: { rules: [] } }) { // @TODO: unify this function between files, maybe even canonize it as a Ruleset/DetachedRuleset method at some point.
 	const map = {}
 
-	// Handle maps that only have one key/value pair (since they will look like a plain list of
-	// length 2).
-	if (list.length === 2 && ! Array.isArray(list[0].value)) {
-		const [{ value: key }, value] = list || [{}]
+	rules.forEach(({ name: key, value: { value } }) => {
+		if (typeof key !== 'string' && key.length === 1 && (key[0] instanceof tree.Keyword)) // Logic adapted from https://github.com/less/less.js/blob/master/lib/less/tree/declaration.js#L46-L49
+			key = key[0].value
 
-		map[key] = value
-	} else
-		list.forEach(({ value: item } = {}) => {
-			if (Array.isArray(item)) {
-				const [{ value: key }, value] = item || [{}]
-
-				map[`${key}`] = value
-			}
-		})
+		map[`${key}`] = value
```

Comment:
I preserved what had existed before before (see removed line 37).

Reply:
Ah, oops 🙃

Reply:
It's still weirdly redundant.

Reply:
Hmm... you’re probably right. I thought I ran an issue when the key was another type, but I can’t remember what it might have been. I’ll try removing and test.

Reply:
Internally, referencing `[key]` will always run `key.toString()` in order to do the lookup, which is exactly what happens when you put it in a template string. I believe this is one of those JS quirks where `[0]` and `['0']` refer to the same key.